scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py

```diff
@@ -59,14 +59,37 @@ def deactivate(self):
         self.view.deactivateButton()
 
     def getWorkspaceNames(self):
-        name = self.view.workspace
+        # get current values
+        original_Re_name = self.view.workspace
+        original_Im_name = self.view.imaginary_workspace
         final_options = self.load.get_workspace_names_for_FFT_analysis(self.view.use_raw_data)
 
+        # update view
         self.view.addItems(final_options)
         self.view.removeRe('PhaseQuad')
         self.removePhaseFromIM(final_options)
 
-        self.view.workspace = name
+        # make intelligent guess of what user wants
+        group_pair_context = self.load.group_pair_context[self.load.group_pair_context.selected]
+        Re_name_to_use = None
+        Im_name_to_use = None
+        default_name = None
+        # will need to check this exists before using it
+        if group_pair_context:
```

Comment:
Is this needed when do we expect the relevant group/pair object to nor exist? Also if this check fails will line 85 below not produce a RuntimeError as it tries to index a None object?

Reply:
if `group_pair_context` does not exist and the current name is not in the list of new names we go to line 84. In this case `default_name` is None so we do not enter the if statement.